### ISSUE TITLE
Documentation for configuration options.

### ISSUE THREAD

**Issue by Sann5** [snap-stanford/GraphGym#52]
Hello! 

This project is truly amazing, thank you. That said I'm finding it difficult to apply it to my own datasets. Naturally, I would like to customize the grid search however, I'm not sure what the valid options are for each field in the configuration. The valid options I know are because of the examples that are provided in the repo, but a comprehensive list for each field would be better. Is there any documentation on this? Im also unsure about how to register my data set.

**Comment by HeVLF** [comment]
Hey! Have you discovered how to register a dataset?

**Comment by Sann5** [comment]
Hey kind of yes. But in the end i had so much trouble using graph gym that i replaced altogether with my own implementation. But if you want i can give you the link to my fork of the torch_geometric graphgym that tries to use it.

**Comment by HeVLF** [comment]
Hey. I'm thinking of doing the same, if the troubles persist. But if you can, please send me. Thank you!

**Comment by Sann5** [comment]
https://github.com/Sann5/pytorch_geometric/tree/costum_dataset_bug

**Comment by HeVLF** [comment]
Thank you. Hopefully I can use it for my study

**Comment by JanineCHEN** [comment]
I fully agree that a comprehensive documentation is really in need for customization. I wonder if this project is still maintained? @JiaxuanYou

**Comment by ChenpengZhang** [comment]
I personally will recommend looking into the graphgym/loader.py file. In the file (specifically the load_pyg, load_nx function) the code is sort of self-explanatory. Looks like the model only supports (PPI, Amazon, Coauthor, KarateClub, MNISTSuperpixels, Planetoid, QM7b, TUDataset) and customized networkx datasets.
